### PR TITLE
Require all tests to follow naming convention

### PR DESCRIPTION
**Pull request by ajtowns**
Based on top of #11774

### CODE REVIEW

**Comment by MarcoFalke** [comment]
utACK 4de3d52e655902a07c85db84f0c6236302ef78a4

**Comment by promag** [comment]
utACK 4de3d52.

**Comment by ajtowns** [comment]
Rebased

**Comment by MarcoFalke** [comment]
Could you also fix the names in `.travis.yml` for the cron job: `--exclude pruning,dbcrash`, please?

File: test/functional/test_runner.py [hunk 1]
```diff
@@ -472,25 +472,15 @@ def was_successful(self):
...
 
     good_prefixes_re = re.compile("(example|feature|interface|mempool|mining|p2p|rpc|wallet)_")
     bad_script_names = [script for script in ALL_SCRIPTS if good_prefixes_re.match(script) is None]
 
-    if len(bad_script_names) < EXPECTED_VIOLATION_COUNT:
-        print("{}HURRAY!{} Number of functional tests violating naming convention reduced!".format(BOLD[1], BOLD[0]))
-        print("Consider reducing EXPECTED_VIOLATION_COUNT from %d to %d" % (EXPECTED_VIOLATION_COUNT, len(bad_script_names)))
-    elif len(bad_script_names) > EXPECTED_VIOLATION_COUNT:
-        print("INFO: %d tests not meeting naming conventions (expected %d):" % (len(bad_script_names), EXPECTED_VIOLATION_COUNT))
+    if len(bad_script_names) > 0:
```
Comment (jnewbery) [inline test/functional/test_runner.py]: more pythonic would be `if bad_script_names:` (https://www.python.org/dev/peps/pep-0008/#id51:  "For sequences, (strings, lists, tuples), use the fact that empty sequences are false.")

File: test/functional/test_runner.py [hunk 2]
```diff
@@ -472,25 +472,15 @@ def was_successful(self):
...
+        print("%sERROR:%s %d tests not meeting naming conventions:" % (BOLD[1], BOLD[0], len(bad_script_names)))
         print("  %s" % ("\n  ".join(sorted(bad_script_names))))
-        assert len(bad_script_names) <= EXPECTED_VIOLATION_COUNT + LEEWAY, "Too many tests not following naming convention! (%d found, expected: <= %d)" % (len(bad_script_names), EXPECTED_VIOLATION_COUNT)
+        assert len(bad_script_names) == 0, "Some tests are not following naming convention!"
```
Comment (jnewbery) [inline test/functional/test_runner.py]: This condition has already been tested. Just raise an assertion error:

```python
raise AssertionError("Some tests are not following naming convention!")
```

**Review by jnewbery — COMMENTED** [review]
Tested ACK 0b94082361d4ffe130766aa14f15dc7144f8239c with a couple of style nits.

Agree with Marco that the travis file needs to be updated.

I think this may need to be rebased to show that only the last commit is new.

**Comment by ajtowns** [comment]
@MarcoFalke ah, sorry about that. #12292 should fix it.

**Comment by MarcoFalke** [comment]
Are you planning on fixing up the nits?

**Comment by ajtowns** [comment]
Nits fixed as suggested, and rebased to make github happier

**Comment by MarcoFalke** [comment]
utACK 125f4a49097062dccc7da2825a4e0c891384f588

**Pull request merged**